assessment/models.py

```diff
@@ -426,3 +421,89 @@ class VideoConferenceRoomSerializer(serializers.ModelSerializer):
     class Meta:
         model = VideoConferenceRoom
         fields = ['room_id', 'conference_in_assessment_event', 'conference_host', 'conference_assessee', 'room_opened', 'conference_participants']
+
+
+class ToolAttempt(PolymorphicModel):
+    tool_attempt_id = models.UUIDField(default=uuid.uuid4, primary_key=True)
+    grade = models.FloatField(default=0)
+    test_flow_attempt = models.ForeignKey('assessment.TestFlowAttempt', on_delete=models.CASCADE)
+    assessment_tool_attempted = models.ForeignKey('assessment.AssessmentTool', on_delete=models.CASCADE, default=None)
+
+
+class AssignmentAttempt(ToolAttempt):
+    file_upload_directory = models.TextField(null=True)
+    filename = models.TextField(default=None, null=True)
```

Comment:
Hello jo, just wondering, will this property be responsible for persisting the originally submitted filename? If so, how do we handle cases where two assesses submitted files with identical names? Let me know :D

Reply:
Hello, Rashad. Our current implementation overwrites the previous submission when a new file is submitted. In doing so, as shown in the assessment_event_attempt service module, both the cloud directory and filename will be updated in accordance to the new submitted file.